### PR TITLE
Added Support for Drawing On Gray Image

### PR DESCRIPTION
**Pull request by legacyshield**

### CODE REVIEW

**Comment by ashu1994** [comment]
i have made necessary changes in paint.go as well and created a pull request for that too. These both changes need to be there for proper running of code.

**Comment by llgcode** [comment]
I've seen the related pull request golang/freetype#41 so I will merge when this one will be merged. If not we can integrate this specific painter in draw2d.
Thanks for your contribution
related to issue #114.

**Comment by ashu1994** [comment]
Yes right...

File: draw2dimg/ftgc.go [hunk 1]
```diff
@@ -58,6 +58,8 @@ func NewGraphicContext(img draw.Image) *GraphicContext {
 	switch selectImage := img.(type) {
 	case *image.RGBA:
 		painter = raster.NewRGBAPainter(selectImage)
+	case *image.Gray:
+		painter = raster.NewGRAYPainter(selectImage)
```
Comment (nigeltao) [inline draw2dimg/ftgc.go]: GRAY should be Gray. The RGBA above is not Rgba as per https://github.com/golang/go/wiki/CodeReviewComments#initialisms but Gray is not an initialism.

Comment (nigeltao) [inline draw2dimg/ftgc.go]: Also, I don't know draw2d very well, but you might want to watch out that NewRGBAPainter uses the draw.Over composition operator, from the image/draw standard library, by default. In comparison, the NewGrayPainter proposed at https://github.com/golang/freetype/pull/41 in effect uses the draw.Src composition operator.

Your GrayPainter's Paint method also ignores the Span.Alpha field, which looks like a mistake.

**Pull request closed without merging**